extensions/amp-social-share/1.0/social-share.js

```diff
@@ -74,12 +65,62 @@ export function SocialShare(props) {
       onClick={() => handleActivation(finalEndpoint, checkedTarget)}
       style={{...size, ...props['style']}}
     >
+      {processChildren(props, size)}
+    </div>
+  );
+}
+
+/**
+ * If the specified type can have children and children exist return children.
+ * Otherwise return the default icon for the specified type.  If the specified
+ * type can have its color / background changed, set those to color /
+ * background specified in props, otherwise use defaults, or do not set if
+ * defaults do not exist.
+ * @param {?PreactDef.Renderable} children
+ * @param {!JsonObject} props
+ * @param {JsonObject} size
+ * @return {PreactDef.Renderable}
+ */
+function processChildren(props, size) {
+  const {
+    'type': type,
+    'children': children,
+    'color': propsColor,
+    'background': propsBackground,
+  } = props;
+  const uType = type.toUpperCase();
+  const typeConfig =
+    getSocialConfig(type) ||
+    dict({
+      'canChangeColor': true,
+      'canHaveChildren': true,
+    });
+
+  if (typeConfig['canHaveChildren'] && children) {
+    return children;
+  } else {
+    const baseStyle = CSS.BASE_STYLE;
+    const iconStyle = dict({
+      'color':
+        typeConfig['canChangeColor'] && propsColor
+          ? propsColor
+          : CSS[uType]
```

Comment:
At this point, having colors in `CSS[uType]` just makes our life harder. Let's move them to the `typeConfig`? The `CSS.BASE_STYLE` can stay where it is, of course.

Reply:
Sounds good, updated!